packages/core/contracts-ovm/insured-bridge/implementation/BridgePool.sol

```diff
@@ -357,23 +368,60 @@ contract BridgePool is Testable, BridgePoolInterface, ExpandedERC20 {
         emit RelayDisputed(depositHash, keccak256(ancillaryData));
     }
 
-    /************************************
-     *           View FUNCTIONS         *
-     ************************************/
+    /**
+     * @notice Synchronize any balance changes in this contract with the utilized & liquid reserves. This would be done
+     * at the conclusion of an L2->L1 token transfer via the canonical token bridge.
```

Comment:
Does the L2 Bridge actually call this method? IIUC it seems that `sync` only gets called indirectly by `exchangeRateCurrent` which gets called by `add/removeLiquidity`?

If so, then I think the `exchangeRateCurrent` name is not as clear as it can be. Some possible changes:
- rename `exchangeRateCurrent` to `syncAndGetCurrentExchangeRate` to signify better that it syncs balance state
- add a `sync` modifier that calls `_sync` and add it to the `exchangeRateCurrent` method, or even the LP functions. 

I think the main confusion for me reading the code is that its not clear what triggers `sync`, but clearly `sync` needs to get called everytime the `liquidReserves` or L1 token balance for this contract changes.

Reply:
This is not called by the L2 bridge method. There is no way to enforce that the conclusion of bridge action from L2->L1 via the canonical bridge are done from a specific contract. We need to think about the finalization of a bridging action as tokens being "dropped" onto the contract at any time. In this framing, the `sync` method can be called at any time to incorporate these balance changes.

The reason I followed this pattern (and named this method as I did) is as follows:
1) `exchangeRateCurrent` is the same name that compound uses for their cToken->underlying token exchange rate logic.
2) There might be a case where you want to `sync` and not do anything else, for example we _might_ want our bot to call `sync` after finalizing an L2->L1 transfer via the bridge.

I do see your point though that it's a bit convoluted that the _only_ time `sync` is called in this contract is via `exchangeRateCurrent`, which is called from the LP methods. The problem with adding it as a modifier is that this is actually pretty gas inefficient. For example, consider adding liquidity. you would: a) sync on the modifier on this method and then sync again on the call to the `exchangerRateCurrent`. The way it's set up now enforces that `sync` is ONLY called when it is needed.

Sync _only_ needs to be called when the L1 token balance for this contract changes, not when adding or removing `liquidReserves`.

I'm happy to rename this method though, as a middle ground. not the _biggest_ fan of `syncAndGetCurrentExchangeRate`. Any other ideas?

Reply:
>exchangeRateCurrent is the same name that compound uses for their cToken->underlying token exchange rate logic.

I think this is a good reason to ease other integrator's onboarding who are used to cTokens.

> The problem with adding it as a modifier is that this is actually pretty gas inefficient. For example, consider adding liquidity. you would: a) sync on the modifier on this method and then sync again on the call to the exchangerRateCurrent. The way it's set up now enforces that sync is ONLY called when it is needed.

Why not remove `sync` from the `exchangeRateCurrent` call, make it a modifier for the LP methods, and then also expose a public `sync` method (that calls `_sync`) for anyone to manually trigger? I'm not sure this is the best pattern IMO and I think its probably fine as is but just throwing an idea out there.